src/script/calling/videoGridHandler.ts

```diff
@@ -0,0 +1,88 @@
+/*
+ * Wire
+ * Copyright (C) 2018 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+import ko from 'knockout';
+import {getDifference} from 'Util/ArrayUtil';
+import {Participant, UserId} from '../calling/Participant';
+
+let baseGrid: string[] = ['', '', '', ''];
+
+export interface Grid {
+  grid: (Participant | null)[];
+  thumbnail: Participant | null;
+  hasRemoteVideo: boolean;
+}
+
+/**
+ * Will compute the next grid layout according to the previous state and the new array of streams
+ * The grid will fill according to this pattern
+ * - 1 stream : [id, '', '', '']
+ * - 2 streams: [id, '', id, '']
+ * - 3 streams: [id, '', id, id]
+ * - 3 streams: [id, id, '', id]
+ * - 4 streams: [id, id, id, id]
+ * @param {Array<string|0>} previousGrid - the previous state of the grid
```

Comment:
JSDoc types are not needed in TypeScript.
```suggestion
 * @param previousGrid the previous state of the grid
```